src/polytopes/ngon.jl

```diff
@@ -58,6 +58,17 @@ innerangles(ngon::Ngon) = innerangles(boundary(ngon))
 
 signarea(ngon::Ngon) = sum(signarea, simplexify(ngon))
 
+"""
+    segments(ngon)
+
+Return the segments linking consecutive points of the `ngon`.
+"""
+function segments(ngon::Ngon)
+  v = vertices(ngon)
+  n = length(v)
+  @inbounds (Segment(v[i], v[mod1(i+1, n)]) for i in 1:n)
+end
```

Comment:
The ngon itself doesn't have segments, its boundary has. So you need to fix the code to use the segments of the boundary of the ngon.